types/mongoose/index.d.ts

```diff
@@ -98,12 +98,15 @@ declare module "mongoose" {
    * which would result in a passing conditional, because `never` is included in all types.
    */
   export type MongooseFilterQuery<T> = {
-    [P in keyof T]?: P extends '_id' ?
-      mongodb.Condition<string | mongodb.ObjectId | { _id: mongodb.ObjectId }> :
-      [Extract<T[P], mongodb.ObjectId>] extends [never] ?
-        mongodb.Condition<T[P]> :
-        mongodb.Condition<T[P] | string>;
-  } & mongodb.RootQuerySelector<T>;
+    [P in keyof T]?: P extends '_id'
+      ? [Extract<T[P], mongodb.ObjectId>] extends [never]
+        ? mongodb.Condition<T[P] | { _id: mongodb.ObjectId }>
```

Comment:
I think this line can simply be `mongodb.Condition<T[P]>`

Reply:
Thanks, I have implemented your suggestion.